### ISSUE TITLE
`unused-void-attr` Linter Check Generates Duplicate and False Positive Errors

### ISSUE THREAD

**Issue by maxonfjvipon** [objectionary/lints#507]
This is `tuple` EO object:
```
# Tuple.
# An ordered, immutable collection of elements.
[prev value length] > tuple
  $ > as-tuple

  # Empty tuple.
  # A tuple with no elements. When dataized, it represents an immutable, empty collection.
  [] > empty
    $ > empty
    $ > as-tuple
    0 > length
    error "Can't get prev from the empty tuple" > prev
    error "Can't get value from the empty tuple" > value

    # Take one element from the tuple, at the given position.
    error "Can't get an object from the empty tuple" > [i] > at

    # Create a new tuple with this element added to the end of it.
    [x] > with
      as-tuple. > @
        tuple
          ^
          x
          (length.plus 1).as-number

  # Take one element from the tuple, at the given position.
  [i] > at
    if. > @
      or.
        0.gt index
        length.lte index
      error "Given index is out of tuple bounds"
      at-fast ^
    i > idx!
    if. > index!
      0.gt idx
      length.plus idx
      idx

    [tup] > at-fast
      if. > @
        (tup.length.plus -1).eq index
        tup.value
        at-fast tup.prev

  # Create a new tuple with this element added to the end of it.
  [x] > with
    as-tuple. > @
      tuple
        ^
        x
        (length.plus 1).as-number
```

After linting I see these errors:
```
<error check="unused-void-attr" line="10" severity="error">The void attribute "prev" is unused anywhere in the object "tuple"</error>
<error check="unused-void-attr" line="10" severity="error">The void attribute "value" is unused anywhere in the object "tuple"</error>
<error check="unused-void-attr" line="10" severity="error">The void attribute "length" is unused anywhere in the object "tuple"</error>
<error check="unused-void-attr" line="10" severity="error">The void attribute "prev" is unused anywhere in the object "tuple"</error>
<error check="unused-void-attr" line="10" severity="error">The void attribute "value" is unused anywhere in the object "tuple"</error>
<error check="unused-void-attr" line="10" severity="error">The void attribute "length" is unused anywhere in the object "tuple"</error>
<error check="unused-void-attr" line="10" severity="error">The void attribute "prev" is unused anywhere in the object "tuple"</error>
<error check="unused-void-attr" line="10" severity="error">The void attribute "value" is unused anywhere in the object "tuple"</error>
<error check="unused-void-attr" line="10" severity="error">The void attribute "length" is unused anywhere in the object "tuple"</error>
<error check="unused-void-attr" line="23" severity="error">The void attribute "i" is unused anywhere in the object "at"</error>
```

1. there are duplicates
2. the `i` attribute is actually used in `at` object, so it's a false positive
3. the lint is weak since all the noticed attributes are actually used "in the object `tuple`"
4. why severity is `error`? Even if we would have the object like: `[a b c] > obj`, there's no error, just unused void attributes. I believe it must be `warning`

**Comment by h1alexbel** [comment]
@maxonfjvipon for `tuple.eo` I getting:

```xml
<defects xmlns:eo="https://www.eolang.org">
   <defect line="3" severity="error">The void attribute "prev" is unused anywhere in the object "tuple"</defect>
   <defect line="3" severity="error">The void attribute "value" is unused anywhere in the object "tuple"</defect>
   <defect line="3" severity="error">The void attribute "length" is unused anywhere in the object "tuple"</defect>
   <defect line="16" severity="error">The void attribute "i" is unused anywhere in the object "at"</defect>
</defects>
```

1. Duplicates seeping from `Program` execution. I think this one should be resolved after #499 gets merged.
2. The attribute `i` inside `at` is marked as unused `i` from this:
  ```eo
  error "Can't get an object from the empty tuple" > [i] > at
  ```
  I'm not sure how we can detect such cases for now.
3. Currently, the lint is designed to pass only voids that used on the object level (not nested). Thus, the void attribute `prev` is not used directly by `tuple`, but used in `empty`. The same for `value` and `length`. We should fully rework the lint to enable this nested object awareness.
4. Agree, it should be a warning. Error seems to be too strict


WDYT?

**Comment by maxonfjvipon** [comment]
@yegor256 please make a minor release
